ibtk/include/ibtk/PETScMatUtilities.h

```diff
@@ -159,18 +130,40 @@ class PETScMatUtilities
                                               int dof_index_idx,
                                               SAMRAI::tbox::Pointer<SAMRAI::hier::PatchLevel<NDIM> > patch_level);
 
-    static inline void ib_4_interp_fcn(const double r, double* const w)
+    static void ib_4_interp_fcn(const double r, double* const w)
     {
-        const double q = sqrt(1.0 + 4.0 * r * (1.0 - r));
-        w[0] = 0.125 * (3.0 - 2.0 * r - q);
-        w[1] = 0.125 * (3.0 - 2.0 * r + q);
-        w[2] = 0.125 * (1.0 + 2.0 * r + q);
-        w[3] = 0.125 * (1.0 + 2.0 * r - q);
+        const double q = sqrt(-7.0 + 12.0 * r - 4.0 * r * r);
+        w[0] = 0.125 * (5.0 - 2.0 * r - q);
+        w[1] = 0.125 * (5.0 - 2.0 * r + q);
+        w[2] = 0.125 * (-1.0 + 2.0 * r + q);
+        w[3] = 0.125 * (-1.0 + 2.0 * r - q);
         return;
     } // ib_4_interp_fcn
 
     static const int ib_4_interp_stencil = 4;
 
+    /*!
+     * \brief Construct a parallel PETSc Mat object corresponding to data
+     * prolongation from a coarser level to a finer level.
+     */
+    static void
+    constructPatchLevelProlongationOp(Mat& mat,
```

Comment:
> On Apr 21, 2015, at 11:47 AM, Amneet Bhalla <notifications@github.com> wrote:
> 
> In ibtk/include/ibtk/PETScMatUtilities.h <https://github.com/IBAMR/IBAMR/pull/6#discussion_r28791527>:
> 
> >          return;
> >      } // ib_4_interp_fcn
> >  
> >      static const int ib_4_interp_stencil = 4;
> >  
> > +    /*!
> > +     * \brief Construct a parallel PETSc Mat object corresponding to data
> > +     * prolongation from a coarser level to a finer level.
> > +     */
> > +    static void
> > +    constructPatchLevelProlongationOp(Mat& mat,
> Makes sense if you think of it as an Op that goes from one level to another, but one can think of it as belonging to the level of origin. Both are fine to me. I can rename it if you prefer...
> 
I don't care which one you use --- I like Op just fine --- the point was that you called a related function XXXMat().

-- Boyce